src/Commands/Self/Plugin/MigrateCommand.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Self\Plugin;
+
+/**
+ * Migrates Terminus 2 plugins.
+ *
+ * @package Pantheon\Terminus\Commands\Self\Plugin
+ */
+class MigrateCommand extends InstallCommand
+{
+    /**
+     * Migrate Terminus 2 plugins.
+     *
+     * @command self:plugin:migrate
+     * @aliases plugin:migrate
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
+     */
+    public function migrate()
```

Comment:
@uberhacker 
Could be optimized a bit (for more verbosity) into something like:

```php
public function migrate(): void
    {
        $plugins_dir = $this->getConfig()->get('plugins2_dir');
        if (!is_dir($plugins_dir)) {
            $this->log()->notice('No Terminus 2 plugins installed: plugins dir is not found.');
            return;
        }

        $plugin_dirs = glob($plugins_dir . DIRECTORY_SEPARATOR . '*', GLOB_ONLYDIR);
        if (!$plugin_dirs) {
            $this->log()->notice('No Terminus 2 plugins installed: plugins not found.');
            return;
        }

        $projects = array_map(
            fn($dir) => $this->getProjectNameFromPath($dir),
            $plugin_dirs
        );
        $this->install($projects);

        $this->log()->notice(
            sprintf('Terminus 2 plugins installed: %s.', implode(', ', $projects))
        );
    }
```

Reply:
This is a good suggestion. In general, nesting should be avoided. Each level of nesting introduces additional "state" that your brain has to keep track of when you are reading and understanding the code. In contrast, if you push the "if" up to the top of the method and exit early with a `return`, as in the suggested review, once you read past the `if`, you can forget about it. It is always an improvement if you can reduce a level of nesting by introducing an early-exit if. Converting a `foreach` into an `array_map` is more hit-and-miss. Sometimes, writing out the foreach longhand is actually better and easier to read. If the logic is very simple, as shown here, then I think the array_map is better.